### PR TITLE
close the uv_async_t handle when the kernel is halted, update example in README

### PR DESCRIPTION
**Pull request by zvin**
Without this node processes never quit.

### CODE REVIEW

File: src/bindings.cc [hunk 1]
```diff
@@ -3,8 +3,6 @@
 #include "disk.h"
```
Comment (petrosagg) [inline src/bindings.cc]: This file no longer needs async.h

File: src/async.cc [hunk 2]
```diff
@@ -25,6 +25,13 @@ void init_async() {
 	uv_mutex_init(&lock);
 }
 
+void close_async() {
+	uv_mutex_destroy(&lock);
+	uv_close((uv_handle_t*)async, [](uv_handle_t* handle) {
```
Comment (petrosagg) [inline src/async.cc]: Why do we have to cast it?

File: src/async.cc [hunk 3]
```diff
@@ -25,6 +25,13 @@ void init_async() {
...
+		free(handle);
```
Comment (petrosagg) [inline src/async.cc]: Not sure about this, but I think in C++ we use `new` and `delete`

File: src/node_lkl.cc [hunk 4]
```diff
@@ -13,11 +14,13 @@ NAN_METHOD(startKernel) {
 
 	lkl_host_ops.print = NULL;
 	int memory = info[0]->Uint32Value();
+	init_async();
```
Comment (petrosagg) [inline src/node_lkl.cc]: Can you add a FIXME comment about guarding the start and halt calls for double calls? Also we are deallocating the async handle that is referenced by the callbacks passed to the user of the lib. We should probably check that async is initialized in `callback_wrapper`

File: src/bindings.cc (hunk 1 above)
Comment (zvin) [inline src/bindings.cc]: fixed

File: src/async.cc (hunk 2 above)
Comment (zvin) [inline src/async.cc]: I get the following error if I don't
```
cannot convert ‘uv_async_t* {aka uv_async_s*}’ to ‘uv_handle_t* {aka uv_handle_s*}’ for argument ‘1’ to ‘void uv_close(uv_handle_t*, uv_close_cb)’
```

File: src/async.cc (hunk 3 above)
Comment (zvin) [inline src/async.cc]: that also works, replaced

File: src/node_lkl.cc (hunk 4 above)
Comment (zvin) [inline src/node_lkl.cc]: double start asserts
```
node: lib/posix-host.c:302: panic: Assertion `0' failed.
```

double stop also asserts
```
node: ../deps/uv/src/unix/core.c:103: uv_close: Assertion `!(handle->flags & (UV_CLOSING | UV_CLOSED))' failed.
```

Comment (zvin) [inline src/node_lkl.cc]: check added in callback_wrapper

Comment (zvin) [inline src/node_lkl.cc]: FIXME comment added

Comment (zvin) [inline src/node_lkl.cc]: On a side note, calling startKernel then haltKernel then startKernel again segfaults.

**Pull request merged**